torch/_subclasses/fake_tensor.py

```diff
@@ -1413,6 +1435,19 @@ def _validate_cache_key(
         # Avoid caching for any ops that would require a more sophisticated
         # caching implementation, e.g., data dependent ops or ops that modify
         # the inputs.
+
+        if isinstance(func, torch._ops.HigherOrderOperator):
+            # For invoke_subgraph op, if the identifier is set then its safe to
+            # cache the fake tensor result.
```

Comment:
this comment is not correct anymore right? It's only safe to cache the result if the validation fn returns True.